src/Database.cpp

```diff
@@ -0,0 +1,460 @@
+#include "Database.hpp"
+#include "contrib/sqlite/sqlite3pp.h"
+#include "contrib/bcrypt/BCrypt.hpp">
+#include "contrib/json11.hpp"
+#include <fstream>
+#include <iostream>
+#include "CNProtocol.hpp"
+#include <string>
+
+#include <regex>
+#include "CNStructs.hpp"
+#include "settings.hpp"
+#include "Player.hpp"
+#include <list>
+#include "CNStructs.hpp"
+
+using namespace json11;
+
+//TODO: replace this sqlite wrapper with something better, clean up queries, get rid of json
+
+sqlite3pp::database db;
+
+void Database::open() {
+	//checking if database file exists
+	std::ifstream file;
+	file.open("data.db");
+	if (file) {
+		file.close();
+		//if exists, assign it		
+		db = sqlite3pp::database ("data.db");
+
+		DEBUGLOG(
+			std::cout << "Database in operation" << std::endl;
+		)
+	}
+	else {
+		//if doesn't, create all the tables
```

Comment:
This is fine for now, but we should probably have a separate table called "DatabaseVersion" or something that keeps track. Something like a primitive database migration system.

Reply:
+1 I second the DatabaseVersion table idea.